core/src/main/java/org/apache/calcite/rel/RelDistributions.java

```diff
@@ -103,7 +103,7 @@ private RelDistributionImpl(Type type, ImmutableIntList keys) {
           || Ordering.natural().isOrdered(keys)
           : "key columns of hash distribution must be in order";
       assert type == Type.HASH_DISTRIBUTED
-          || type == Type.RANDOM_DISTRIBUTED
+          || type == Type.RANGE_DISTRIBUTED
           || keys.isEmpty();
     }
```

Comment:
@rendel Thank you for reporting this. I think it is a bug. Could you file a JIRA for this?

Reply:
I would say a test case is more important than JIRA.
If the assertion is user-facing, then it should be a regular if + illegalargumentexception